### DISCUSSION TITLE
Getting the data from the db

### DISCUSSION THREAD

**Discussion by TinusJ** [AsifArmanRahman/firebase-rest-api#4] [Q&A]
Hi, how would i get the data 

This is what i have 
```
u = get_db().child(f"users/{u['localId']}").get(u.get('idToken'))
print(u.val())
```

**Comment by AsifArmanRahman** [comment]
Hi  @TinusJ, what is the output printed by that ``print`` function.

**Comment by TinusJ** [comment]
If was printing just a None, and previously without the .val it was printing the obj Firebaseresponse.

On a side note i found the issue where it was not finding my document on the db, how can I do a where id='someValue' ?

This is how the data is structured.

![image](https://user-images.githubusercontent.com/16685359/180877984-f5956be7-c99f-4279-a743-6cb7ccdcc919.png)

something similar to this
self.get_db().child("users").where("id",isEqualTo = u['localId']).get(u.get('idToken'))

**Reply by AsifArmanRahman** [reply]
First of all, the FirebaseResponse object is actually an dict object, which you can break using a ``for`` loop

```python
u_info = get_db().child(f"users/{u['localId']}").get(u.get('idToken'))

for info in u_info:
    print(info.key(), ': ', info.val())
````

<br>

***

Secondly,

> found the issue where it was not finding my document on the db

 From the seems of it, you're using the below code to save data to your db.
```python
get_db().child("users").push(data, u.get('idToken))
```

thus it's creating an timestamp id and storing the data inside it.
![image](https://user-images.githubusercontent.com/60575508/180913678-ed7c0501-32d7-4f52-b895-1101fcac638c.png)

If you opt to do it this way

```python
get_db().child(f"users/{u['localId']}").set(data, u.get('idToken))
```

Which would give you an architecture of
```
-   users:
    - user_local_id:
        - user_data1: 'data1'
        - user_data2: 'data2'
```

This way you can use the code I've written on the very first of this comment, and no need to store the localId value then search for it.

<br>
***
Finally for this

> something similar to this
> self.get_db().child("users").where("id",isEqualTo = u['localId']).get(u.get('idToken'))

Check out the code snippet [here](https://firebase-rest-api.readthedocs.io/en/latest/guide/database.html#equal-to).

**Reply by TinusJ** [reply]
Ah thanks a lot :), didn't find the docs but now I have it .

**Reply by AsifArmanRahman** [reply]
Link to the docs is added in the right panel of the GitHub Repo, in **About** section. And if you meant docs to the services (authentication, database, storage) they are under **Integrate Firebase>>Use Services**. I will add links in README and put the services on the navbar eventually.
